src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(action, value) {
+      action.call(this, value);
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(value) {
+      this.result += value;
+    },
+
+    subtract(value) {
+      this.result -= value;
+    },
+
+    multiply(value) {
+      this.result *= value;
+    },
+
+    divide(value) {
+      this.result = value === 0 ? 0 : this.result / value;
+    },
```

Comment:
Mathematically speaking, [division by zero is undefined](https://en.wikipedia.org/wiki/Division_by_zero), i.e. definitely not `0`. `NaN` is a better option. 

Alternatively, since js does not throw an error on division by zero, you can just not handle DbZ at all.

Reply:
Agree, but I want to avoid any errors in this case. Ideally, no matter what number the user types, the application continues to work